### PR TITLE
Update FAQ with new features, how to run image, and how to parameterize

### LINKED ISSUE #70: missing documentation on how to run built image
it would be nice to be able to run my application from the plugin or at least instructions on how to do so after my build.

### LINKED ISSUE #349: Provide documentation about how to parameterize at runtime.
# JVM Flags

`docker run -e "JAVA_TOOL_OPTIONS=<JVM flags>" <image name>`

or kubernetes:

```yaml
apiVersion: v1
kind: Pod
spec:
  containers:
  - name: <name>
    image: <image>
    env:
    - name: JAVA_TOOL_OPTIONS
      value: <JVM flags>
```

# Environment

`docker run -e "NAME=VALUE" <image name>`

or kubernetes:

```yaml
apiVersion: v1
kind: Pod
spec:
  containers:
  - name: <name>
    image: <image>
    env:
    - name: MY_NAME
      value: is q chen
    - name: ASK_QUESTION
      value: who ${MY_NAME}
```

# Arguments to main

`docker run <image name> <arg1> <arg2> <arg3>`

or kubernetes:

```yaml
apiVersion: v1
kind: Pod
spec:
  containers:
  - name: <name>
    image: <image>
    args:
    - <arg1>
    - <arg2>
    - <arg3>
```

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #70, fixes #349

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -22,6 +23,70 @@ If a question you have is not answered below, please [submit an issue](/../../is
 
 See [rules_docker](https://github.com/bazelbuild/rules_docker) for a similar existing container image build tool for the [Bazel build system](https://github.com/bazelbuild/bazel). The tool can build images for languages such as Python, NodeJS, Java, Scala, Groovy, C, Go, Rust, and D.
 
+### How do I run the image I built?
+
+If you built your image directly to the Docker daemon using `jib:dockerBuild` (Maven) or `jibDockerBuild` (Gradle), you simply need to use `docker run <image name>`.
+
+If you built your image to a registry using `jib:build` (Maven) or `jib` (Gradle), you will need to pull the image using `docker pull <image name>` before using `docker run`.
+
+To run your image on Kubernetes, see [steps 4-6 of the Kubernetes Engine deployment tutorial](https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app#step_4_create_a_container_cluster).
```
Comment (coollog) [inline docs/faq.md]: I think we can provide the kubectl command here like in https://github.com/coollog/micronaut-jib#with-kubernetes

File: docs/faq.md [hunk 2]
```diff
@@ -98,11 +155,12 @@ We currently support adding a custom directory with an **incubating** feature. T
...
-
-You can also [`docker pull`](https://docs.docker.com/engine/reference/commandline/pull/) the image built with Jib to have it available in your local Docker daemon.
+There are several ways of doing this:
 
-You can also [run a local Docker registry](https://docs.docker.com/registry/deploying/) and point Jib to push to the local registry.
+- Use [`jib:dockerBuild` for Maven](../jib-maven-plugin#build-to-docker-daemon) or [`jibDockerBuild` for Gradle](../jib-gradle-plugin#build-to-docker-daemon) to build directly to your local Docker daemon.
+- Use [`jib:buildTar` for Maven]() or [`jibBuildTar` for Gradle]() to build the image to a tarball, then use `docker load --input` to load the image into Docker (the tarball built with these commands will be located in `target/jib-image.tar` for Maven and `build/jib-image.tar` for Gradle by default).
+- [`docker pull`](https://docs.docker.com/engine/reference/commandline/pull/) the image built with Jib to have it available in your local Docker daemon.
+- [Run a local Docker registry](https://docs.docker.com/registry/deploying/) and point Jib to push to the local registry.
```
Comment (coollog) [inline docs/faq.md]: Maybe prefix with: "Alternatively, instead of using a Docker daemon, you can ..."

File: docs/faq.md [hunk 3]
```diff
@@ -22,6 +23,70 @@ If a question you have is not answered below, please [submit an issue](/../../is
...
+If you built your image directly to the Docker daemon using `jib:dockerBuild` (Maven) or `jibDockerBuild` (Gradle), you simply need to use `docker run <image name>`.
+
+If you built your image to a registry using `jib:build` (Maven) or `jib` (Gradle), you will need to pull the image using `docker pull <image name>` before using `docker run`.
+
+To run your image on Kubernetes, see [steps 4-6 of the Kubernetes Engine deployment tutorial](https://cloud.google.com/kubernetes-engine/docs/tutorials/hello-app#step_4_create_a_container_cluster).
+
+### How do I set parameters for my image at runtime?
+
+#### JVM Flags
+
```
Comment (briandealwis) [inline docs/faq.md]: I think it's necessary to say this applies for `distroless/java`.  Other JRE images may require using other environment variables.

File: docs/faq.md [hunk 4]
```diff
@@ -98,11 +155,12 @@ We currently support adding a custom directory with an **incubating** feature. T
...
 ### Can I build to a local Docker daemon?
 
-See [`jib:dockerBuild` for Maven](../jib-maven-plugin#build-to-docker-daemon) and [`jibDockerBuild` for Gradle](../jib-gradle-plugin#build-to-docker-daemon).
+There are several ways of doing this:
 
-You can also [`docker pull`](https://docs.docker.com/engine/reference/commandline/pull/) the image built with Jib to have it available in your local Docker daemon.
-
-You can also [run a local Docker registry](https://docs.docker.com/registry/deploying/) and point Jib to push to the local registry.
+- Use [`jib:dockerBuild` for Maven](../jib-maven-plugin#build-to-docker-daemon) or [`jibDockerBuild` for Gradle](../jib-gradle-plugin#build-to-docker-daemon) to build directly to your local Docker daemon.
+- Use [`jib:buildTar` for Maven]() or [`jibBuildTar` for Gradle]() to build the image to a tarball, then use `docker load --input` to load the image into Docker (the tarball built with these commands will be located in `target/jib-image.tar` for Maven and `build/jib-image.tar` for Gradle by default).
```
Comment (briandealwis) [inline docs/faq.md]: Missing links

File: docs/faq.md (hunk 2 above)
Comment (briandealwis) [reply docs/faq.md]: _Run a local container registry, such as a Docker registry or other repository managers._?  There's Nexus (open-source), Artifactory, …

File: docs/faq.md [hunk 5]
```diff
@@ -22,6 +23,74 @@ If a question you have is not answered below, please [submit an issue](/../../is
...
+If you built your image directly to the Docker daemon using `jib:dockerBuild` (Maven) or `jibDockerBuild` (Gradle), you simply need to use `docker run <image name>`.
+
+If you built your image to a registry using `jib:build` (Maven) or `jib` (Gradle), you will need to pull the image using `docker pull <image name>` before using `docker run`.
+
+To run your image on Kubernetes, you can use kubectl:
+
+```shell
+kubectl run jib-deployment --image=<image name>
+```
+
```
Comment (coollog) [inline docs/faq.md]: I think also having the reference you had initially to the tutorial would be good.

File: docs/faq.md [hunk 6]
```diff
@@ -98,11 +159,12 @@ We currently support adding a custom directory with an **incubating** feature. T
...
-
-You can also [`docker pull`](https://docs.docker.com/engine/reference/commandline/pull/) the image built with Jib to have it available in your local Docker daemon.
+There are several ways of doing this:
 
-You can also [run a local Docker registry](https://docs.docker.com/registry/deploying/) and point Jib to push to the local registry.
+- Use [`jib:dockerBuild` for Maven](../jib-maven-plugin#build-to-docker-daemon) or [`jibDockerBuild` for Gradle](../jib-gradle-plugin#build-to-docker-daemon) to build directly to your local Docker daemon.
+- Use [`jib:buildTar` for Maven]() or [`jibBuildTar` for Gradle]() to build the image to a tarball, then use `docker load --input` to load the image into Docker (the tarball built with these commands will be located in `target/jib-image.tar` for Maven and `build/jib-image.tar` for Gradle by default).
+- [`docker pull`](https://docs.docker.com/engine/reference/commandline/pull/) the image built with Jib to have it available in your local Docker daemon.
+- Alternatively, instead of using a Docker daemon, you can [run a local Docker registry](https://docs.docker.com/registry/deploying/) and point Jib to push to the local registry.
```
Comment (briandealwis) [inline docs/faq.md]: _run a local registry, such as a Docker registry, and point …_

(there are other implementations which may be better for their needs, like Nexus, Artifactory, etc.)

File: docs/faq.md (hunk 3 above)
Comment (chanseokoh) [reply docs/faq.md]: Interesting. I thought this variable is natively supported (`https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars002.html`), but not really?

Comment (chanseokoh) [reply docs/faq.md]: Not that I've ever used or tested it.

Comment (TadCordle) [reply docs/faq.md]: One of the links in there leads here: https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#tooloptions

which says "Platforms which support environment variables or other named strings, may support the JAVA_TOOL_OPTIONS variable." So I'm not sure it's guaranteed to work.

Comment (briandealwis) [reply docs/faq.md]: https://stackoverflow.com/questions/28327620/difference-between-java-options-java-tool-options-and-java-opts
  - `_JAVA_OPTIONS` trumps `JAVA_TOOL_OPTIONS`, though it's OpenJDK/Oracle specific
  - IBM J9 and OpenJ9 support `IBM_JAVA_OPTIONS`
  - there's also a `JAVA_OPTS` that isn't supported directly by the JVM but is used by a lot of start scripts.

File: docs/faq.md (hunk 5 above)
Comment (briandealwis) [reply docs/faq.md]: and maybe put `jib-deployment` into `<jib-deployment>`?

**Pull request merged**